# Case Study_Week06

#### Data Set used : StudentsPerformance
#### 1000 rows × 8 columns

In [1]:
import pandas as pd
import numpy as np

In [2]:
data=pd.read_csv('StudentsPerformance.csv')
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

# 1. Find out how many males and females participated in the test

In [3]:
data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

# 2. What do you think about the students' parental level of education

In [4]:
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

#### Insights :

#### The highest count of 226 is for the category "some college" and the least count of 59 is for the category "master's degree".

# 3. Who scores the most on average for math, reading and writing based on
#### ● Gender
#### ● Test preparation course

In [5]:
data.rename(columns={'math score':'math_score', 'reading score':'reading_score', 'writing score':'writing_score'}, inplace=True)

In [6]:
data.groupby('gender', as_index=False)[['math_score', 'reading_score', 'writing_score']].mean()

gender  math_score  reading_score  writing_score
0  female   63.376448      72.590734      72.467181
1    male   68.821577      65.545643      63.446058

##### Insights based on Gender :

#### Highest average for math - male
#### Highest average for reading - female
#### Highest average for writing - female

In [7]:
data.groupby('test preparation course', as_index=False)[['math_score', 'reading_score', 'writing_score']].mean()

test preparation course  math_score  reading_score  writing_score
0               completed    69.96648      74.175978      74.684358
1                    none    63.78972      66.417445      64.457944

##### Insights based on Test preparation course :

#### Those who completed test preparation course scored the highest average for math, reading and writing.

# 4. What do you think about the scoring variation for math, reading and writing based on
#### ● Gender
#### ● Test preparation course

In [8]:
data.groupby('gender', as_index=False)[['math_score', 'reading_score', 'writing_score']].var()

gender  math_score  reading_score  writing_score
0  female  256.958593     207.677438     220.369327
1    male  211.889097     200.211010     202.413924

In [9]:
data.groupby('gender', as_index=False)[['math_score', 'reading_score', 'writing_score']].std()

gender  math_score  reading_score  writing_score
0  female   16.029928      14.411018      14.844842
1    male   14.556411      14.149594      14.227225

##### Insights based on Gender :

#### Variance : The least variation in scoring for math, reading and writing is shown by male when compared to that of female.

#### Standard Deviation  :The least standard deviation in scoring for math, reading and writing is shown by male when compared to that of female.

In [10]:
data.groupby('test preparation course', as_index=False)[['math_score', 'reading_score', 'writing_score']].var()

test preparation course  math_score  reading_score  writing_score
0               completed  210.884027     183.265864     175.202612
1                    none  246.668662     213.419851     226.251739

In [11]:
data.groupby('test preparation course', as_index=False)[['math_score', 'reading_score', 'writing_score']].std()

test preparation course  math_score  reading_score  writing_score
0               completed   14.521847      13.537572      13.236412
1                    none   15.705689      14.608896      15.041667

##### Insights based on Test preparation course :

#### Variance : Those who completed test preparation course shows the least variation in scoring for math, reading and writing.

#### Standard Deviation : Those who completed test preparation course shows the least standard deviation in scoring for math, reading and writing.

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

In [12]:
data['math_score'].quantile(0.75)

77.0

In [13]:
top=data[data.math_score > np.quantile(data.math_score,0.75)]
top

gender race/ethnicity parental level of education         lunch  \
430  female        group A                some college      standard   
431  female        group B           bachelor's degree  free/reduced   
432  female        group D           bachelor's degree      standard   
433  female        group D                 high school      standard   
434  female        group D           bachelor's degree  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math_score  reading_score  writing_score  
430               completed          78             87             91  
431                    none          78             79             76  
432                    none          78             82             79  
433                    none          78             81             80  
434                    none          78             90             93  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[231 rows x 8 columns]

#### Insights :

#### The above data shows the top 25% of the students based on their maths score and this will help the management to give bonus points to these students.